### PR TITLE
types: add version string to block header

### LINKED ISSUE #495: Add version string to block header
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):

Feature Request (by ebuchman)

For 0.10 tendermint release

Add one string to the block header with a version string, which is set when we compile the software.  This string will be equal between two nodes iff they are running the same version of the software.  This will allow us to much more easily coordinate upgrades across nodes.

This may contain two parts, like API version enabled, and API version activated

### PR DESCRIPTION
**Pull request by zramsay**
- closes #495, #723

### CODE REVIEW

File: types/block.go [hunk 1]
```diff
@@ -202,6 +206,7 @@ func (h *Header) StringIndented(indent string) string {
 %s  Data:           %v
 %s  Validators:     %v
 %s  App:            %v
+%s  Version:	    %v
```
Comment (melekes) [inline types/block.go]: formatting

Comment (zramsay) [reply types/block.go]: hrm, looks ok here https://github.com/tendermint/tendermint/blob/495-version-in-block-header/types/block.go#L209 and also locally:
screenshot from 2017-10-12 08-27-56

am I missing something?

Comment (odeke-em) [reply types/block.go]: @zramsay you have a tab in there, if my assumption is that you are using vim in Ubuntu, type `:set list` and you'll see the tab right after `Version:^I`, of
which we aren't using tabs in the entire format string, please remove it and manually line it up with spaces. Once done, `:set nolist` to return to normal.

Comment (zramsay) [reply types/block.go]: thanks, fixed

**Comment by melekes** [comment]
Don't forget to update the CHANGELOG

**Comment by odeke-em** [comment]
@melekes he updated the changelog, we are ready for flight! Thanks @zramsay!

**Comment by ebuchman** [comment]
I don't want to merge this just yet until there is some discussion on the use of this variable. We need to have a larger discussion about how to roll out upgrades

**Comment by ebuchman** [comment]
Thanks Zach, but gonna close pending further discussion

**Pull request closed without merging**